SS2-Project/Assets/Starstorm2/Modules/Pickups/Equipments/AffixPurple.cs

```diff
@@ -1,22 +1,171 @@
-﻿using RoR2;
+﻿using MSU;
+using R2API;
+using RoR2;
+using RoR2.ContentManagement;
+using RoR2.Projectile;
+using System.Collections;
 using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using static MSU.BaseBuffBehaviour;
+
 namespace SS2.Equipments
 {
 #if DEBUG
-    /*public class AffixPurple : SS2EliteEquipment
+    public class AffixPurple : SS2EliteEquipment
     {
-        public override EquipmentDef EquipmentDef { get; } = SS2Assets.LoadAsset<EquipmentDef>("ElitePurpleEquipment", SS2Bundle.Indev);
+        //public override EquipmentDef EquipmentDef { get; } = SS2Assets.LoadAsset<EquipmentDef>("ElitePurpleEquipment", SS2Bundle.Indev);
+
+        //public override List<MSEliteDef> EliteDefs { get; } = new List<MSEliteDef>
+        //{
+        //    SS2Assets.LoadAsset<MSEliteDef>("edPurple", SS2Bundle.Indev),
+        //    SS2Assets.LoadAsset<MSEliteDef>("edPurpleHonor", SS2Bundle.Indev)
+        //};
+
+        //public override bool FireAction(EquipmentSlot slot)
+        //{
+        //    return false;
+        //}
+
+        public override List<EliteDef> EliteDefs => _eliteDefs;
+        private List<EliteDef> _eliteDefs;
+
+        public override NullableRef<List<GameObject>> ItemDisplayPrefabs => null;
+
+        public override EquipmentDef EquipmentDef => _equipmentDef;
+        private EquipmentDef _equipmentDef;
+
+        public static DotController.DotIndex index;
+
+        public BuffDef _buffPurplePoisonBuildup; // TODO: { get; } = SS2Assets.LoadAsset<BuffDef>("bdPoisonBuildup", SS2Bundle.Indev);
+        public BuffDef _buffPurplePoisonDebuff; // { get; } = SS2Assets.LoadAsset<BuffDef>("bdPurplePoison", SS2Bundle.Indev);
+        public BuffDef _buffAffixPurple; //{ get; } = SS2Assets.LoadAsset<BuffDef>("bdElitePurple", SS2Bundle.Indev);
+
+        public override void Initialize()
+        {
+            // TODO: Do I need a separate dot for poisondebuff buffdef? idk
+            index = DotAPI.RegisterDotDef(0.25f, 0.18f, DamageColorIndex.DeathMark, _buffPurplePoisonBuildup);
+        }
 
-        public override List<MSEliteDef> EliteDefs { get; } = new List<MSEliteDef>
+        public override bool IsAvailable(ContentPack contentPack)
         {
-            SS2Assets.LoadAsset<MSEliteDef>("edPurple", SS2Bundle.Indev),
-            SS2Assets.LoadAsset<MSEliteDef>("edPurpleHonor", SS2Bundle.Indev)
-        };
+            return false;
+        }
+
+        public override IEnumerator LoadContentAsync()
+        {
+            yield break;
+        }
 
-        public override bool FireAction(EquipmentSlot slot)
+        public override bool Execute(EquipmentSlot slot)
         {
             return false;
         }
-    }*/
+
+        public override void OnEquipmentLost(CharacterBody body)
+        {
+        }
+
+        public override void OnEquipmentObtained(CharacterBody body)
+        {
+        }
+
+        public class PurplePoisonBuildup : BaseBuffBehaviour
```

Comment:
Remove class for now as it contains nothing

Reply:
actually its because I havent finished implementing the poison elite yet, you build up stacks of "poison" and once it reaches X stacks, you begin taking damage. 

For some reason I thought I had it working but guess not...

Reply:
ah wait I see what you meant, yeah we dont need the classes, just the buffdef objects in the editor